src/test/java/com/aws/greengrass/shadowmanager/util/JsonUtilTest.java

```diff
@@ -96,4 +96,21 @@ void GIVEN_no_source_node_and_bad_update_node_WHEN_validatePayload_THEN_throws_i
         assertThat(thrown.getErrorMessage().getErrorCode(), is(400));
         assertThat(thrown.getErrorMessage().getMessage(), is("Invalid version"));
     }
+
+    @Test
+    void GIVEN_state_with_6_levels_WHEN_validatePayload_THEN_successfully_validates() throws IOException {
+        ShadowDocument source = new ShadowDocument();
+        JsonNode updateNode = getPayloadJson("{\"state\": {\"desired\": {\"1\": {\"2\": {\"3\": {\"4\": {\"5\": {\"6\": \"The Beatles\"}}}}}}}}".getBytes()).get();
+        assertDoesNotThrow(() -> JsonUtil.validatePayload(source, updateNode));
```

Comment:
can we be extra thorough here and do two checks, one with `desired` and one with `reported`?